music_assistant/server/providers/deezer/helpers.py

```diff
@@ -0,0 +1,576 @@
+        return result
+
+    return await asyncio.to_thread(_search)
+
+
+async def get_access_token(mass, app_id, app_secret, code) -> Credential:
+    """Update the access_token."""
+    response = await _post_http(
+        mass=mass,
+        url="https://connect.deezer.com/oauth/access_token.php",
+        data={
+            "code": code,
+            "app_id": app_id,
+            "secret": app_secret,
+        },
+        params={
+            "code": code,
+            "app_id": app_id,
+            "secret": app_secret,
+        },
+        headers=None,
+    )
+    try:
+        return response.split("=")[1].split("&")[0]
+    except Exception as error:
+        raise LoginFailed("Invalid auth code") from error
+
+
+async def _post_http(mass, url, data, params=None, headers=None) -> str:
+    async with mass.http_session.post(
+        url, headers=headers, params=params, json=data, ssl=False
+    ) as response:
+        if response.status != 200:
+            raise ConnectionError(f"HTTP Error {response.status}: {response.reason}")
+        response_text = await response.text()
+        return response_text
+
+
+def parse_artist(mass, artist: deezer.Artist) -> Artist:
+    """Parse the deezer-python artist to a MASS artist."""
+    return Artist(
+        item_id=str(artist.id),
+        provider=mass.domain,
+        name=artist.name,
+        media_type=MediaType.ARTIST,
+        provider_mappings={
+            ProviderMapping(
+                item_id=str(artist.id),
+                provider_domain=mass.domain,
+                provider_instance=mass.instance_id,
+            )
+        },
+        metadata=parse_metadata_artist(artist=artist),
+    )
+
+
+async def parse_album(mass, album: deezer.Album) -> Album:
+    """Parse the deezer-python album to a MASS album."""
+    return Album(
+        album_type=AlbumType(album.type),
+        item_id=str(album.id),
+        provider=mass.domain,
+        name=album.title,
+        artists=[parse_artist(mass=mass, artist=album.get_artist())],
+        media_type=MediaType.ALBUM,
+        provider_mappings={
+            ProviderMapping(
+                item_id=str(album.id),
+                provider_domain=mass.domain,
+                provider_instance=mass.instance_id,
+            )
+        },
+        metadata=parse_metadata_album(album=album),
+    )
+
+
+async def parse_playlist(mass, playlist: deezer.Playlist) -> Playlist:
+    """Parse the deezer-python playlist to a MASS playlist."""
+    return Playlist(
+        item_id=str(playlist.id),
+        provider=mass.domain,
+        name=playlist.title,
+        media_type=MediaType.PLAYLIST,
+        provider_mappings={
+            ProviderMapping(
+                item_id=str(playlist.id),
+                provider_domain=mass.domain,
+                provider_instance=mass.instance_id,
+            )
+        },
+        metadata=await parse_metadata_playlist(playlist=playlist),
```

Comment:
```suggestion
        metadata.images.append(....
```

Also, are you sure the image is always there ?

Reply:
Yes, Deezer auto-generates pictures for playlists :slightly_smiling_face: